### PR TITLE
#446 - Upgrade CLI to 2024-10.

### PR DESCRIPTION
**Pull request by Thels**
For now this is only ensuring that the draft version is updated - not that the examples/functionality exposed by the upgraded CALM version would provide - these are covered by other issues.

### CODE REVIEW

**Comment by rocketstack-matt** [comment]
Is the version in shared/package.json expected to be kept in sync with the cli? What about the workspace version? Do we need / want to adopt a convention here?

**Review by rocketstack-matt — CHANGES_REQUESTED** [review]
version should be updated in [cli/src/index.ts](https://github.com/finos/architecture-as-code/blob/c0fa2d76301598141843c589e814689c23d28f26/cli/src/index.ts#L20)

**Comment by Thels** [comment]
> Is the version in shared/package.json expected to be kept in sync with the cli? What about the workspace version? Do we need / want to adopt a convention here?

It doesn't need to be kept in sync -the version within `shared` is effectively moot as it's a non-published package.

File: shared/src/commands/validate/validate.spec.ts [hunk 1]
```diff
@@ -369,7 +369,7 @@ describe('validate-all', () => {
                     'error',
                     'must be integer',
                     '/nodes/0/interfaces/0/port',
-                    'https://raw.githubusercontent.com/finos/architecture-as-code/main/calm/draft/2024-04/meta/interface.json#/defs/host-port-interface/properties/port/type'
+                    'https://raw.githubusercontent.com/finos/architecture-as-code/main/calm/draft/2024-10/meta/interface.json#/defs/host-port-interface/properties/port/type'
```
[the remark this answers is not in the document]
Comment (Thels) [reply shared/src/commands/validate/validate.spec.ts:372]: Applied both - good spot.

**Comment by Thels** [comment]
> version should be updated in [cli/src/index.ts](https://github.com/finos/architecture-as-code/blob/c0fa2d76301598141843c589e814689c23d28f26/cli/src/index.ts#L20)

Good spot - I've included resolving it directly from the `package.json` - as keeping this one up to date manually would be a pain.

**Comment by rocketstack-matt** [comment]
> > version should be updated in [cli/src/index.ts](https://github.com/finos/architecture-as-code/blob/c0fa2d76301598141843c589e814689c23d28f26/cli/src/index.ts#L20)
> 
> Good spot - I've included resolving it directly from the `package.json` - as keeping this one up to date manually would be a pain.

Unless you've figured out a new way, this won't work - see this thread - https://github.com/finos/architecture-as-code/issues/283

**Comment by Thels** [comment]
> > > version should be updated in [cli/src/index.ts](https://github.com/finos/architecture-as-code/blob/c0fa2d76301598141843c589e814689c23d28f26/cli/src/index.ts#L20)
> > 
> > 
> > Good spot - I've included resolving it directly from the `package.json` - as keeping this one up to date manually would be a pain.
> 
> Unless you've figured out a new way, this won't work - see this thread - #283

I have! It's included in both the TS config and the bundler is handling it for us.

It inlines `var version = "0.3.0";` into the disted artifact that is published/ran as the CLI.

**Comment by rocketstack-matt** [comment]
> > > > version should be updated in [cli/src/index.ts](https://github.com/finos/architecture-as-code/blob/c0fa2d76301598141843c589e814689c23d28f26/cli/src/index.ts#L20)
> > > 
> > > 
> > > Good spot - I've included resolving it directly from the `package.json` - as keeping this one up to date manually would be a pain.
> > 
> > 
> > Unless you've figured out a new way, this won't work - see this thread - #283
> 
> I have! It's included in both the TS config and the bundler is handling it for us.
> 
> It inlines `var version = "0.3.0";` into the disted artifact that is published/ran as the CLI.

Awesome!

**Review by rocketstack-matt — APPROVED** [review]

**Pull request merged**